MalApi/MyAnimeListApi.cs

```diff
@@ -72,6 +78,40 @@ private HttpRequestMessage InitNewRequest(string uri, HttpMethod method)
             return request;
         }
 
+        private HttpRequestMessage InitNewRequestWithCredentials(string user, string password, string uri, HttpMethod method, UpdateObjectBase updateInfo = null)
+        {
+            HttpRequestMessage request = new HttpRequestMessage(method, uri);
+
+            // Requests with credentials require them to be encoded in base64
+            string credentials;
+            if (user != null && password != null)
+            {
+                var plainTextBytes = Encoding.UTF8.GetBytes(user + ":" + password);
+                credentials =  Convert.ToBase64String(plainTextBytes);
+            }
+            else
+            {
+                throw new Exception("Invalid input credentials format (both username and password have to be input OR just the base64 representation");
+            }
+
+            if (UserAgent != null)
+            {
+                request.Headers.TryAddWithoutValidation("User-Agent", UserAgent);
+                // Adding the authorization header with the credentials
+                request.Headers.TryAddWithoutValidation("Authorization", "Basic " + credentials);
+            }
+
+            // Encoding and adding the new information in the content(body) of the request
+			if(updateInfo != null)
+			{
+				var xml = updateInfo.GenerateXml();
+				var content = "data=" + WebUtility.UrlEncode(xml);
+				request.Content = new StringContent(content, Encoding.UTF8, "application/x-www-form-urlencoded");
```

Comment:
Does this not work? The FormUrlEncodedContent class exists so you don't have to encode it yourself.

```
request.Content = new FormUrlEncodedContent(new KeyValuePair<string, string>[] { new KeyValuePair<string, string>("data", xml) });
```